File: queencitysocial/src/predicthq_api.js
// Load environment variables from the .env file
require('dotenv').config();
const axios = require('axios');

// Define the API endpoint and your API key
const url = "https://api.predicthq.com/v1/events/";
const apiKey = process.env.API_KEY;

const currentDate = new Date().toISOString().split('T')[0];
// Define the query parameters
const params = {
    within: "15mi@35.2271,-80.8431",   // Charlotte, NC latitude and longitude
    "active.gte": currentDate,        // Start date
    "active.tz": "America/New_York",   // Time zone
    sort: "start"
    };

// Set up the headers, including authorization
const headers = {
    Authorization: `Bearer ${apiKey}`,
    Accept: "application/json"
};

// Function to fetch events
async function fetchEvents() {
    try {
        const response = await axios.get(url, { headers, params });
        console.log(response.data); // Print or process the JSON response
    } catch (error) {
        console.error(`Error: ${error.response.status} - ${error.response.statusText}`);
    }
}

fetchEvents();

File: queencitysocial/python/predicthq-api.py
import requests

# Define the API endpoint and your API key
url = "https://api.predicthq.com/v1/events/"
api_key = "YOUR_API_KEY"  # Replace with your PredictHQ API key

# Define the query parameters
params = {
    "within": "15mi@35.2271,-80.8431",  # Charlotte, NC latitude and longitude
    "active.gte": "2024-01-01",          # Start date
    "active.lte": "2024-12-31",          # End date
    "active.tz": "America/New_York",     # Time zone
    "limit": 50,                         # Number of results to return
    "category": "community,festivals,concerts"  # Example categories
}

# Set up the headers, including authorization
headers = {
    "Authorization": f"Bearer {api_key}",
    "Accept": "application/json"
}

# Send the request to PredictHQ
response = requests.get(url, headers=headers, params=params)

# Check if the request was successful
if response.status_code == 200:
    # Print or process the JSON response
    events = response.json()
    print(events)
else:
    print(f"Error: {response.status_code} - {response.text}")
